lucene/codecs/src/java/org/apache/lucene/codecs/uniformsplit/IntersectBlockReader.java

```diff
@@ -291,23 +293,263 @@ public void seekExact(BytesRef term, TermState state) {
     throw new UnsupportedOperationException();
   }
 
-  //TODO this approach is a hack to reuse complex code in AutomatonTermsEnum.  How to make ATE's logic more reusable?
-  protected static class AutomatonNextTermCalculator extends AutomatonTermsEnum {
+  //This is a copy of AutomatonTermsEnum.  Since it's an inner class, the outer class can
```

Comment:
@jpountz as you asked, I forked AutomatonTermsEnum into UniformSplit's IntersectBlockReader, and undid the minor change in ATE.